url0e7d57473a_kyorohiro_HetimaUtil_tgz-pJ8-net_hetimatan_net_http_HttpServer3xxJ8/src/net/hetimatan/util/http/HttpObjectHelper.java

```diff
@@ -0,0 +1,54 @@
+package net.hetimatan.util.http;
+
+import net.hetimatan.net.http.PieceInfoList;
+
+public class HttpObjectHelper {
+	private static long[] sdef = {0, Long.MAX_VALUE};
+
+	public static PieceInfoList getRangeList(String range, long max) {
+		long[] v= getRange(range, max);
+		PieceInfoList list = new PieceInfoList();
+		for(int i=0;i+1<v.length;i+=2) {
+			list.append(v[i], v[i+1]);
+		}
+		return list;
+	}
+
+	public static long[] getRange(String range) {
+		return getRange(range, Long.MAX_VALUE);
+	}
+
+	public static long[] getRange(String range,long max) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 34-49 into a separate method.